weave

```diff
@@ -1277,10 +1277,16 @@ check_overlap() {
 # with_container_addresses.
 ipam_reclaim() {
     CONTAINER_ID="$1"
-    # The weave IP addresses of containers attached by the plugin are
-    # recorded specially in IPAM, since the container id is not know
-    # at the time.
+
+    # Weave keeps track of which container has which IP in its IPAM.
+    # If the container's network interface is not the same as Weave's
+    # then it is likely the container was not started by Weave.
```

Comment:
This isn't quite right; how it's started is orthogonal to how it gets attached to the network.

"if the network interface is not named `ethwe`, then it was attached via Docker Plugin or via CNI"